Index: src/main/java/ru/javawebinar/topjava/SpringMain.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/SpringMain.java	(revision )
+++ src/main/java/ru/javawebinar/topjava/SpringMain.java	(revision )
@@ -0,0 +1,23 @@
+package ru.javawebinar.topjava;
+
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import ru.javawebinar.topjava.repository.UserRepository;
+
+import java.util.Arrays;
+
+/**
+ * User: gkislin
+ * Date: 22.08.2014
+ */
+public class SpringMain {
+    public static void main(String[] args) {
+        ConfigurableApplicationContext appCtx = new ClassPathXmlApplicationContext("spring/spring-app.xml");
+        System.out.println(Arrays.toString(appCtx.getBeanDefinitionNames()));
+
+//        UserRepository userRepository = (UserRepository) appCtx.getBean("mockUserRepository");
+        UserRepository userRepository = appCtx.getBean(UserRepository.class);
+        userRepository.getAll();
+        appCtx.close();
+    }
+}
Index: src/main/java/ru/javawebinar/topjava/repository/mock/MockUserRepositoryImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/repository/mock/MockUserRepositoryImpl.java	(revision )
+++ src/main/java/ru/javawebinar/topjava/repository/mock/MockUserRepositoryImpl.java	(revision )
@@ -0,0 +1,47 @@
+package ru.javawebinar.topjava.repository.mock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.javawebinar.topjava.model.User;
+import ru.javawebinar.topjava.repository.UserRepository;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * GKislin
+ * 15.06.2015.
+ */
+public class MockUserRepositoryImpl implements UserRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(MockUserRepositoryImpl.class);
+
+    @Override
+    public boolean delete(int id) {
+        LOG.info("delete " + id);
+        return true;
+    }
+
+    @Override
+    public User save(User user) {
+        LOG.info("save " + user);
+        return user;
+    }
+
+    @Override
+    public User get(int id) {
+        LOG.info("get " + id);
+        return null;
+    }
+
+    @Override
+    public List<User> getAll() {
+        LOG.info("getAll");
+        return Collections.emptyList();
+    }
+
+    @Override
+    public User getByEmail(String email) {
+        LOG.info("getByEmail " + email);
+        return null;
+    }
+}
Index: src/main/resources/spring/spring-app.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/spring/spring-app.xml	(revision )
+++ src/main/resources/spring/spring-app.xml	(revision )
@@ -0,0 +1,8 @@
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+
+    <bean class="ru.javawebinar.topjava.repository.mock.MockUserRepositoryImpl" id="mockUserRepository"/>
+
+</beans>
\ No newline at end of file
Index: src/main/java/ru/javawebinar/topjava/web/MealServlet.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/web/MealServlet.java	(date 1465331514000)
+++ src/main/java/ru/javawebinar/topjava/web/MealServlet.java	(revision )
@@ -3,7 +3,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import ru.javawebinar.topjava.model.UserMeal;
-import ru.javawebinar.topjava.repository.InMemoryUserMealRepositoryImpl;
+import ru.javawebinar.topjava.repository.mock.InMemoryUserMealRepositoryImpl;
 import ru.javawebinar.topjava.repository.UserMealRepository;
 import ru.javawebinar.topjava.util.UserMealsUtil;
 
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- pom.xml	(date 1465331514000)
+++ pom.xml	(revision )
@@ -16,6 +16,8 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
+        <spring.version>4.2.6.RELEASE</spring.version>
+
         <!-- Logging -->
         <logback.version>1.1.7</logback.version>
         <slf4j.version>1.7.21</slf4j.version>
@@ -67,6 +69,18 @@
             <scope>runtime</scope>
         </dependency>
 
+        <!-- Spring -->
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>${spring.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
         <dependency>
             <groupId>javax.servlet</groupId>
             <artifactId>jstl</artifactId>
Index: src/main/java/ru/javawebinar/topjava/repository/InMemoryUserMealRepositoryImpl.java
===================================================================
--- src/main/java/ru/javawebinar/topjava/repository/InMemoryUserMealRepositoryImpl.java	(date 1465331514000)
+++ src/main/java/ru/javawebinar/topjava/repository/mock/InMemoryUserMealRepositoryImpl.java	(revision )
@@ -1,6 +1,7 @@
-package ru.javawebinar.topjava.repository;
+package ru.javawebinar.topjava.repository.mock;
 
 import ru.javawebinar.topjava.model.UserMeal;
+import ru.javawebinar.topjava.repository.UserMealRepository;
 import ru.javawebinar.topjava.util.UserMealsUtil;
 
 import java.util.Collection;
