Index: src/main/resources/spring/spring-app.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/spring/spring-app.xml	(date 1465332050000)
+++ src/main/resources/spring/spring-app.xml	(revision )
@@ -1,8 +1,10 @@
 <beans xmlns="http://www.springframework.org/schema/beans"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-       xsi:schemaLocation="
-        http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
 
     <bean class="ru.javawebinar.topjava.repository.mock.MockUserRepositoryImpl" id="mockUserRepository"/>
 
+    <bean class="ru.javawebinar.topjava.service.UserServiceImpl">
+        <property name="repository" ref="mockUserRepository"/>
+    </bean>
 </beans>
\ No newline at end of file
Index: src/main/java/ru/javawebinar/topjava/SpringMain.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/SpringMain.java	(date 1465332050000)
+++ src/main/java/ru/javawebinar/topjava/SpringMain.java	(revision )
@@ -2,7 +2,10 @@
 
 import org.springframework.context.ConfigurableApplicationContext;
 import org.springframework.context.support.ClassPathXmlApplicationContext;
+import ru.javawebinar.topjava.model.Role;
+import ru.javawebinar.topjava.model.User;
 import ru.javawebinar.topjava.repository.UserRepository;
+import ru.javawebinar.topjava.service.UserService;
 
 import java.util.Arrays;
 
@@ -18,6 +21,10 @@
 //        UserRepository userRepository = (UserRepository) appCtx.getBean("mockUserRepository");
         UserRepository userRepository = appCtx.getBean(UserRepository.class);
         userRepository.getAll();
+
+        UserService userService = appCtx.getBean(UserService.class);
+        System.out.println(userService.save(new User(1, "userName", "email", "password", Role.ROLE_ADMIN)));
+
         appCtx.close();
     }
 }
Index: src/main/java/ru/javawebinar/topjava/service/UserServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/service/UserServiceImpl.java	(date 1465332050000)
+++ src/main/java/ru/javawebinar/topjava/service/UserServiceImpl.java	(revision )
@@ -15,6 +15,10 @@
 
     private UserRepository repository;
 
+    public void setRepository(UserRepository repository) {
+        this.repository = repository;
+    }
+
     public User save(User user) {
         return repository.save(user);
     }
