Index: src/main/java/ru/javawebinar/topjava/model/UserMeal.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/model/UserMeal.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/model/UserMeal.java	(revision )
@@ -6,8 +6,7 @@
  * GKislin
  * 11.01.2015.
  */
-public class UserMeal {
-    private Integer id;
+public class UserMeal extends BaseEntity {
 
     private final LocalDateTime dateTime;
 
@@ -20,16 +19,12 @@
     }
 
     public UserMeal(Integer id, LocalDateTime dateTime, String description, int calories) {
-        this.id = id;
+        super(id);
         this.dateTime = dateTime;
         this.description = description;
         this.calories = calories;
     }
 
-    public void setId(int id) {
-        this.id = id;
-    }
-
     public LocalDateTime getDateTime() {
         return dateTime;
     }
@@ -40,14 +35,6 @@
 
     public int getCalories() {
         return calories;
-    }
-
-    public Integer getId() {
-        return id;
-    }
-
-    public boolean isNew() {
-        return id == null;
     }
 
     @Override
Index: src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java
===================================================================
--- src/main/java/ru/javawebinar/topjava/model/UserMealWithExceed.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/to/UserMealWithExceed.java	(revision )
@@ -1,4 +1,4 @@
-package ru.javawebinar.topjava.model;
+package ru.javawebinar.topjava.to;
 
 import java.time.LocalDateTime;
 
Index: src/main/java/ru/javawebinar/topjava/util/TimeUtil.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/util/TimeUtil.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/util/TimeUtil.java	(revision )
@@ -1,5 +1,6 @@
 package ru.javawebinar.topjava.util;
 
+import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 
@@ -9,6 +10,9 @@
  */
 public class TimeUtil {
     private static final DateTimeFormatter DATE_TME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    public static final LocalDate MIN_DATE = LocalDate.of(1, 1, 1);
+    public static final LocalDate MAX_DATE = LocalDate.of(3000, 1, 1);
 
     public static <T extends Comparable<? super T>> boolean isBetween(T value, T start, T end) {
         return value.compareTo(start) >= 0 && value.compareTo(end) <= 0;
Index: src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/util/UserMealsUtil.java	(revision )
@@ -1,7 +1,7 @@
 package ru.javawebinar.topjava.util;
 
 import ru.javawebinar.topjava.model.UserMeal;
-import ru.javawebinar.topjava.model.UserMealWithExceed;
+import ru.javawebinar.topjava.to.UserMealWithExceed;
 
 import java.time.LocalDate;
 import java.time.LocalDateTime;
\ No newline at end of file
Index: src/main/java/ru/javawebinar/topjava/service/UserMealServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/service/UserMealServiceImpl.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/service/UserMealServiceImpl.java	(revision )
@@ -1,13 +1,51 @@
 package ru.javawebinar.topjava.service;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ru.javawebinar.topjava.model.UserMeal;
 import ru.javawebinar.topjava.repository.UserMealRepository;
+import ru.javawebinar.topjava.util.exception.ExceptionUtil;
 
+import java.time.LocalDateTime;
+import java.util.Collection;
+
 /**
  * GKislin
  * 06.03.2015.
  */
+@Service
 public class UserMealServiceImpl implements UserMealService {
 
+    @Autowired
     private UserMealRepository repository;
 
+    @Override
+    public UserMeal get(int id, int userId) {
+        return ExceptionUtil.checkNotFoundWithId(repository.get(id, userId), id);
+    }
+
+    @Override
+    public void delete(int id, int userId) {
+        ExceptionUtil.checkNotFoundWithId(repository.delete(id, userId), id);
+    }
+
+    @Override
+    public Collection<UserMeal> getBetweenDateTimes(LocalDateTime startDateTime, LocalDateTime endDateTime, int userId) {
+        return repository.getBetween(startDateTime, endDateTime, userId);
+    }
+
+    @Override
+    public Collection<UserMeal> getAll(int userId) {
+        return repository.getAll(userId);
+    }
+
+    @Override
+    public UserMeal update(UserMeal meal, int userId) {
+        return ExceptionUtil.checkNotFoundWithId(repository.save(meal, userId), meal.getId());
+    }
+
+    @Override
+    public UserMeal save(UserMeal meal, int userId) {
+        return repository.save(meal, userId);
+    }
 }
Index: src/main/java/ru/javawebinar/topjava/SpringMain.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/SpringMain.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/SpringMain.java	(revision )
@@ -4,9 +4,15 @@
 import org.springframework.context.support.ClassPathXmlApplicationContext;
 import ru.javawebinar.topjava.model.Role;
 import ru.javawebinar.topjava.model.User;
+import ru.javawebinar.topjava.to.UserMealWithExceed;
+import ru.javawebinar.topjava.web.meal.UserMealRestController;
 import ru.javawebinar.topjava.web.user.AdminRestController;
 
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.Month;
 import java.util.Arrays;
+import java.util.List;
 
 /**
  * User: gkislin
@@ -19,6 +25,14 @@
             System.out.println(Arrays.toString(appCtx.getBeanDefinitionNames()));
             AdminRestController adminUserController = appCtx.getBean(AdminRestController.class);
             System.out.println(adminUserController.create(new User(1, "userName", "email", "password", Role.ROLE_ADMIN)));
+            System.out.println();
+
+            UserMealRestController mealController = appCtx.getBean(UserMealRestController.class);
+            List<UserMealWithExceed> filteredMealsWithExceeded =
+                    mealController.getBetween(
+                            LocalDate.of(2015, Month.MAY, 30), LocalTime.of(7, 0),
+                            LocalDate.of(2015, Month.MAY, 31), LocalTime.of(11, 0));
+            filteredMealsWithExceeded.forEach(System.out::println);
         }
     }
 }
Index: src/main/java/ru/javawebinar/topjava/service/UserMealService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/service/UserMealService.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/service/UserMealService.java	(revision )
@@ -1,8 +1,31 @@
 package ru.javawebinar.topjava.service;
 
+import ru.javawebinar.topjava.model.UserMeal;
+import ru.javawebinar.topjava.util.exception.NotFoundException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Collection;
+
 /**
  * GKislin
  * 15.06.2015.
  */
 public interface UserMealService {
+    UserMeal get(int id, int userId) throws NotFoundException;
+
+    void delete(int id, int userId) throws NotFoundException;
+
+    default Collection<UserMeal> getBetweenDates(LocalDate startDate, LocalDate endDate, int userId) {
+        return getBetweenDateTimes(LocalDateTime.of(startDate, LocalTime.MIN), LocalDateTime.of(endDate, LocalTime.MAX), userId);
+    }
+
+    Collection<UserMeal> getBetweenDateTimes(LocalDateTime startDateTime, LocalDateTime endDateTime, int userId);
+
+    Collection<UserMeal> getAll(int userId);
+
+    UserMeal update(UserMeal meal, int userId) throws NotFoundException;
+
+    UserMeal save(UserMeal meal, int userId);
 }
Index: src/main/java/ru/javawebinar/topjava/web/meal/UserMealRestController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/web/meal/UserMealRestController.java	(date 1465849318000)
+++ src/main/java/ru/javawebinar/topjava/web/meal/UserMealRestController.java	(revision )
@@ -1,12 +1,71 @@
 package ru.javawebinar.topjava.web.meal;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import ru.javawebinar.topjava.LoggedUser;
+import ru.javawebinar.topjava.model.UserMeal;
 import ru.javawebinar.topjava.service.UserMealService;
+import ru.javawebinar.topjava.to.UserMealWithExceed;
+import ru.javawebinar.topjava.util.TimeUtil;
+import ru.javawebinar.topjava.util.UserMealsUtil;
 
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+
 /**
  * GKislin
  * 06.03.2015.
  */
+@Controller
 public class UserMealRestController {
+    private static final Logger LOG = LoggerFactory.getLogger(UserMealRestController.class);
+
+    @Autowired
     private UserMealService service;
 
+    public UserMeal get(int id) {
+        int userId = LoggedUser.id();
+        LOG.info("get meal {} for User {}", id, userId);
+        return service.get(id, userId);
+    }
+
+    public void delete(int id) {
+        int userId = LoggedUser.id();
+        LOG.info("delete meal {} for User {}", id, userId);
+        service.delete(id, userId);
+    }
+
+    public List<UserMealWithExceed> getAll() {
+        int userId = LoggedUser.id();
+        LOG.info("getAll for User {}", userId);
+        return UserMealsUtil.getWithExceeded(service.getAll(userId), LoggedUser.getCaloriesPerDay());
+    }
+
+    public void update(UserMeal meal, int id) {
+        meal.setId(id);
+        int userId = LoggedUser.id();
+        LOG.info("update {} for User {}", meal, userId);
+        service.update(meal, userId);
+    }
+
+    public UserMeal create(UserMeal meal) {
+        meal.setId(null);
+        int userId = LoggedUser.id();
+        LOG.info("create {} for User {}", meal, userId);
+        return service.save(meal, userId);
+    }
+
+    public List<UserMealWithExceed> getBetween(LocalDate startDate, LocalTime startTime, LocalDate endDate, LocalTime endTime) {
+        int userId = LoggedUser.id();
+        LOG.info("getBetween dates {} - {} for time {} - {} for User {}", startDate, endDate, startTime, endTime, userId);
+
+        return UserMealsUtil.getFilteredWithExceeded(
+                service.getBetweenDates(
+                        startDate != null ? startDate : TimeUtil.MIN_DATE, endDate != null ? endDate : TimeUtil.MAX_DATE, userId),
+                        startTime != null ? startTime : LocalTime.MIN, endTime != null ? endTime : LocalTime.MAX, LoggedUser.getCaloriesPerDay()
+        );
+    }
-}
+}
\ No newline at end of file
Index: src/main/webapp/mealList.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/mealList.jsp	(date 1465849318000)
+++ src/main/webapp/mealList.jsp	(revision )
@@ -32,7 +32,7 @@
         </tr>
         </thead>
         <c:forEach items="${mealList}" var="meal">
-            <jsp:useBean id="meal" scope="page" type="ru.javawebinar.topjava.model.UserMealWithExceed"/>
+            <jsp:useBean id="meal" scope="page" type="ru.javawebinar.topjava.to.UserMealWithExceed"/>
             <tr class="${meal.exceed ? 'exceeded' : 'normal'}">
                 <td>
                         <%--${meal.dateTime.toLocalDate()} ${meal.dateTime.toLocalTime()}--%>
\ No newline at end of file
