Index: src/main/webapp/mealEdit.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/mealEdit.jsp	(date 1465332464000)
+++ src/main/webapp/mealEdit.jsp	(revision )
@@ -25,8 +25,8 @@
 </head>
 <body>
 <section>
-    <h2><a href="">Home</a></h2>
-    <h3>Edit meal</h3>
+    <h2><a href="index.html">Home</a></h2>
+    <h3>${param.action == 'create' ? 'Create meal' : 'Edit meal'}</h3>
     <hr>
     <jsp:useBean id="meal" type="ru.javawebinar.topjava.model.UserMeal" scope="request"/>
     <form method="post" action="meals">
