Index: src/main/webapp/index.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/index.jsp	(revision )
+++ src/main/webapp/index.jsp	(revision )
@@ -0,0 +1,26 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+
+<fmt:setBundle basename="messages.app"/>
+
+<html>
+<jsp:include page="fragments/headTag.jsp"/>
+<body>
+<jsp:include page="fragments/bodyHeader.jsp"/>
+<section>
+    <form method="post" action="users">
+    Login as: <select name="userId">
+    <option value="100000" selected>User</option>
+    <option value="100001">Admin</option>
+</select>
+    <button type="submit">Выбрать</button>
+</form>
+<ul>
+    <li><a href="users">User List</a></li>
+    <li><a href="meals">Meal List</a></li>
+</ul>
+</section>
+<jsp:include page="fragments/footer.jsp"/>
+</body>
+</html>
\ No newline at end of file
Index: src/main/resources/messages/app_ru.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>NATIVE_TO_ASCII_UTF-8
===================================================================
--- src/main/resources/messages/app_ru.properties	(revision )
+++ src/main/resources/messages/app_ru.properties	(revision )
@@ -0,0 +1,4 @@
+app.title=Подсчет калорий
+app.footer=Приложение по проекту <a href\="https\://github.com/JavaOPs/topjava" target\=_blank>Maven/ Spring/ Security/ JPA(Hibernate)/ Jackson/jQuery</a>
+
+users.title=Список пользователей
\ No newline at end of file
Index: src/main/webapp/fragments/headTag.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/fragments/headTag.jsp	(revision )
+++ src/main/webapp/fragments/headTag.jsp	(revision )
@@ -0,0 +1,10 @@
+<%@page contentType="text/html" pageEncoding="UTF-8" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jstl/core" %>
+<fmt:setBundle basename="messages.app"/>
+
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    <title><fmt:message key="app.title"/></title>
+    <link rel="stylesheet" href="css/style.css">
+</head>
\ No newline at end of file
Index: src/main/webapp/fragments/bodyHeader.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/fragments/bodyHeader.jsp	(revision )
+++ src/main/webapp/fragments/bodyHeader.jsp	(revision )
@@ -0,0 +1,5 @@
+<%@page contentType="text/html" pageEncoding="UTF-8" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+
+<fmt:setBundle basename="messages.app"/>
+<header><a href="meals"><fmt:message key="app.title"/></a></header>
\ No newline at end of file
Index: src/main/webapp/fragments/footer.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/fragments/footer.jsp	(revision )
+++ src/main/webapp/fragments/footer.jsp	(revision )
@@ -0,0 +1,5 @@
+<%@page contentType="text/html" pageEncoding="UTF-8" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<fmt:setBundle basename="messages.app"/>
+<hr>
+<footer><fmt:message key="app.footer"/></footer>
\ No newline at end of file
Index: src/main/resources/messages/app.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>NATIVE_TO_ASCII_UTF-8
===================================================================
--- src/main/resources/messages/app.properties	(revision )
+++ src/main/resources/messages/app.properties	(revision )
@@ -0,0 +1,4 @@
+app.title=Calories management
+app.footer=Project <a href\="https\://github.com/JavaOPs/topjava" target\=_blank>Maven/ Spring/ Security/ JPA(Hibernate)/ Jackson/jQuery</a> application
+
+users.title=User list
\ No newline at end of file
Index: src/main/webapp/userList.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/userList.jsp	(date 1467809722000)
+++ src/main/webapp/userList.jsp	(revision )
@@ -1,10 +1,40 @@
 <%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+
+<fmt:setBundle basename="messages.app"/>
+
 <html>
-<head>
-    <title>User list</title>
-</head>
+<jsp:include page="fragments/headTag.jsp"/>
 <body>
-<h2><a href="index.html">Home</a></h2>
-<h2>User list</h2>
+<jsp:include page="fragments/bodyHeader.jsp"/>
+
+<section>
+    <h3><fmt:message key="users.title"/></h3>
+
+    <table border="1" cellpadding="8" cellspacing="0">
+        <thead>
+        <tr>
+            <th>Name</th>
+            <th>Email</th>
+            <th>Roles</th>
+            <th>Active</th>
+            <th>Registered</th>
+        </tr>
+        </thead>
+        <c:forEach items="${userList}" var="user">
+            <jsp:useBean id="user" scope="page" type="ru.javawebinar.topjava.model.User"/>
+            <tr>
+                <td><c:out value="${user.name}"/></td>
+                <td><a href="mailto:${user.email}">${user.email}</a></td>
+                <td>${user.roles}</td>
+                <td><%=user.isEnabled()%>
+                </td>
+                <td><fmt:formatDate value="${user.registered}" pattern="dd-MMMM-yyyy"/></td>
+            </tr>
+        </c:forEach>
+    </table>
+</section>
+<jsp:include page="fragments/footer.jsp"/>
 </body>
 </html>
Index: src/main/java/ru/javawebinar/topjava/web/UserServlet.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/web/UserServlet.java	(date 1467809722000)
+++ src/main/java/ru/javawebinar/topjava/web/UserServlet.java	(revision )
@@ -1,8 +1,12 @@
 package ru.javawebinar.topjava.web;
 
 import org.slf4j.Logger;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.context.support.WebApplicationContextUtils;
 import ru.javawebinar.topjava.AuthorizedUser;
+import ru.javawebinar.topjava.web.user.AdminRestController;
 
+import javax.servlet.ServletConfig;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
@@ -18,6 +22,15 @@
 public class UserServlet extends HttpServlet {
     private static final Logger LOG = getLogger(UserServlet.class);
 
+    private AdminRestController adminController;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        WebApplicationContext springContext = WebApplicationContextUtils.getRequiredWebApplicationContext(getServletContext());
+        adminController = springContext.getBean(AdminRestController.class);
+    }
+
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
         int userId = Integer.valueOf(request.getParameter("userId"));
         AuthorizedUser.setId(userId);
@@ -25,7 +38,9 @@
     }
 
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        LOG.debug("redirect to userList");
-        response.sendRedirect("userList.jsp");
+        LOG.debug("getAll");
+        request.setAttribute("userList", adminController.getAll());
+        request.getRequestDispatcher("/userList.jsp").forward(request, response);
+//        response.sendRedirect("userList.jsp");
     }
 }
Index: src/main/webapp/index.html
===================================================================
--- src/main/webapp/index.html	(date 1467809722000)
+++ src/main/webapp/index.html	(date 1467809722000)
@@ -1,19 +0,0 @@
-<!DOCTYPE html>
-<html>
-<head>
-    <meta charset="UTF-8">
-    <title>Java Enterprise (Topjava)</title>
-</head>
-<body>
-<h2>Проект "<a href="https://github.com/JavaWebinar/topjava07" target="_blank">Java Enterprise (Topjava)"</a></h2>
-<hr>
-<form method="post" action="users">
-    <b>Meal list of</b>
-    <select name="userId">
-        <option value="100000">User</option>
-        <option value="100001">Admin</option>
-    </select>
-    <button type="submit">Выбрать</button>
-</form>
-</body>
-</html>
Index: src/main/webapp/mealList.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/mealList.jsp	(date 1467809722000)
+++ src/main/webapp/mealList.jsp	(revision )
@@ -9,7 +9,7 @@
 </head>
 <body>
 <section>
-    <h2><a href="index.html">Home</a></h2>
+    <h2><a href="index.jsp">Home</a></h2>
     <h3>Meal list</h3>
     <form method="post" action="meals?action=filter">
         <dl>
\ No newline at end of file
Index: src/main/webapp/css/style.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/css/style.css	(date 1467809722000)
+++ src/main/webapp/css/style.css	(revision )
@@ -22,3 +22,11 @@
 .exceeded {
     color: red;
 }
+
+header, footer {
+    background: none repeat scroll 0 0 #A6C9E2;
+    color: #2E6E9E;
+    font-size: 20px;
+    padding: 5px 20px;
+    margin: 6px 0;
+}
\ No newline at end of file
